odoo/addons/base/tests/test_reports.py

```diff
@@ -12,24 +12,42 @@
 @odoo.tests.tagged('post_install', '-at_install', 'post_install_l10n')
 class TestReports(odoo.tests.TransactionCase):
     def test_reports(self):
-        domain = [('report_type', 'like', 'qweb')]
-        for report in self.env['ir.actions.report'].search(domain):
-            report_model = 'report.%s' % report.report_name
-            try:
-                self.env[report_model]
-            except KeyError:
-                # Only test the generic reports here
-                _logger.info("testing report %s", report.report_name)
-                report_model = self.env[report.model]
-                report_records = report_model.search([], limit=10)
-                if not report_records:
-                    _logger.info("no record found skipping report %s", report.report_name)
+        common_domain = [('report_type', 'like', 'qweb')]
+        to_test = [
+            (
+                [('move_id.move_type', 'in', ('in_invoice', 'in_receipt'))],
+                [('report_name', '=', 'account.report_original_vendor_bill')],
+            ),
+            (
+                [('move_id.move_type', 'in', ('out_invoice', 'out_refund', 'out_receipt', 'in_invoice', 'in_refund', 'in_receipt'))],
+                [('report_name', 'in', ('account.report_invoice_with_payments', 'account.report_invoice'))],
+            ),
+            (
+                [],
+                [],
+            ),
+        ]
+        to_exclude_ids = set()
+        for report_model_domain, report_domain in to_test:
+            full_report_domain = common_domain + report_domain + [('id', 'not in', tuple(to_exclude_ids))]
+            for report in self.env['ir.actions.report'].search(full_report_domain):
+                to_exclude_ids.add(report.id)
+                report_model = 'report.%s' % report.report_name
+                try:
+                    self.env[report_model]
+                except KeyError:
+                    # Only test the generic reports here
+                    _logger.info("testing report %s", report.report_name)
+                    report_model = self.env[report.model]
+                    report_records = report_model.search(report_model_domain, limit=10)
+                    if not report_records:
+                        _logger.info("no record found skipping report %s", report.report_name)
```

Comment:
actually, I think it would be less messy if we do something like
```python
        specific_model_domains = {
            'account.report_original_vendor_bill': [('move_id.move_type', 'in', ('in_invoice', 'in_receipt'))],
            'account.report_invoice_with_payments': [('move_id.move_type', 'in', ('out_invoice', 'out_refund', 'out_receipt', 
'in_invoice', 'in_refund', 'in_receipt'))],
            'account.report_invoice': [('move_id.move_type', 'in', ('out_invoice', 'out_refund', 'out_receipt', 'in_invoice', 'in_refund', 'in_receipt'))],
        }    
        domain = [('report_type', 'like', 'qweb')]
        for report in self.env['ir.actions.report'].search(domain):
            report_model = 'report.%s' % report.report_name
            try:
                self.env[report_model]
            except KeyError:
                # Only test the generic reports here
                _logger.info("testing report %s", report.report_name)
                report_model = self.env[report.model]
                obj_domain = specific_model_domains.get(report.report_name, [])
                report_records = report_model.search(obj_domain, limit=10)
                if not report_records:
                    _logger.info("no record found skipping report %s", report.report_name)
```

so the diff would be limited to 1 line + the definition of specific_model_domains.

:pray: sorry if I misguided in that direction